wandb/artifacts.py

```diff
@@ -230,6 +231,94 @@ def remap_entry(entry):
+                # 'secondaryName:alias' notation.
+                idx = alias.index(":")
+                artifact_collection_name = alias[:idx-1]
+                tag = alias[idx+1:]
+            else:
+                artifact_collection_name = name
+                tag = alias
+            alias_specs.append({
+                'artifactCollectionName': artifact_collection_name,
+                'alias': tag,
+            })
+
+        """Returns the server artifact."""
+        self._server_artifact = self._api.create_artifact(
+            type,
+            name,
+            self._digest,
+            metadata=metadata,
+            aliases=alias_specs, labels=labels, description=description,
+            is_user_created=self._is_user_created)
+        # TODO(artifacts):
+        #   if it's committed, all is good. If it's committing, just moving ahead isn't necessarily
+        #   correct. It may be better to poll until it's committed or failed, and then decided what to
+        #   do
+        artifact_id = self._server_artifact['id']
+        if self._server_artifact['state'] == 'COMMITTED' or self._server_artifact['state'] == 'COMMITTING':
+            # TODO: update aliases, labels, description etc?
+            if use_after_commit:
+                self._api.use_artifact(artifact_id)
+            return self._server_artifact
+        elif self._server_artifact['state'] != 'PENDING':
+            # TODO: what to do in this case?
+            raise Exception('Server artifact not in PENDING state', self._server_artifact)
+
+        # Upload Artifact "L0" files. This should only be artifact_manifest.json. We need to use
+        # the use_prepare_flow, so that the file entry is created in our database before the
+        # upload to cloud storage commences
+        for path, hash, local_path in self._server_manifest_entries:
+            # We need to use the "use_prepare_flow" option
+            self._file_pusher.file_changed(path, local_path, artifact_id, use_prepare_flow=True)
+
+        step_prepare = file_pusher.step_prepare.StepPrepare(self._api, 0.1, 0.01, 1000)  # TODO: params
+        step_prepare.start()
+
+        # Upload Artifact "L1" files, the actual artifact contents
+        self._file_pusher.store_manifest_files(
+            self._manifest,
+            artifact_id,
+            lambda entry, progress_callback : self._manifest.storage_policy.store_file(
+                artifact_id,
+                entry,
+                step_prepare,
+                progress_callback=progress_callback))
+
+        def on_commit():
+            if use_after_commit:
+                self._api.use_artifact(artifact_id)
+            step_prepare.shutdown()
+
+        # This will queue the commit. It will only happen after all the file uploads are done
+        self._file_pusher.commit_artifact(artifact_id, on_commit=on_commit)
+        return self._server_artifact
+
+    def wait(self):
```

Comment:
Heh. Do we still need this one? I thought we'd figured out a workaround. Maybe not...